cmd/keeper/keeper.go

```diff
@@ -1610,37 +1633,57 @@ func keeper(cmd *cobra.Command, args []string) {
 		die("cannot create data dir: %v", err)
 	}
 
+	if _, ok := validAuthMethods[cfg.pgReplAuthMethod]; !ok {
+		die("--pg-repl-auth-method must be one of: md5, trust")
+	}
 	if cfg.pgReplUsername == "" {
 		die("--pg-repl-username is required")
 	}
-
-	if cfg.pgReplPassword == "" && cfg.pgReplPasswordFile == "" {
+	if cfg.pgReplAuthMethod == "trust" {
+		stdout("warning: not utilizing a password for replication between hosts is extremely dangerous")
+		if cfg.pgReplPassword != "" || cfg.pgReplPasswordFile != "" {
+			die("can not utilize --pg-repl-auth-method trust together with --pg-repl-password or --pg-repl-passwordfile")
+		}
+	}
+	if cfg.pgSUAuthMethod == "trust" {
+		stdout("warning: not utilizing a password for superuser is extremely dangerous")
+		if cfg.pgSUPassword != "" || cfg.pgSUPasswordFile != "" {
+			die("can not utilize --pg-su-auth-method trust together with --pg-su-password or --pg-su-passwordfile")
+		}
+	}
+	if cfg.pgReplAuthMethod != "trust" && cfg.pgReplPassword == "" && cfg.pgReplPasswordFile == "" {
 		die("one of --pg-repl-password or --pg-repl-passwordfile is required")
 	}
-	if cfg.pgReplPassword != "" && cfg.pgReplPasswordFile != "" {
+	if cfg.pgReplAuthMethod != "trust" && cfg.pgReplPassword != "" && cfg.pgReplPasswordFile != "" {
 		die("only one of --pg-repl-password or --pg-repl-passwordfile must be provided")
 	}
-	if cfg.pgSUPassword == "" && cfg.pgSUPasswordFile == "" {
+	if _, ok := validAuthMethods[cfg.pgSUAuthMethod]; !ok {
+		die("--pg-su-auth-method must be one of: md5, password, trust")
+	}
+	if cfg.pgSUAuthMethod != "trust" && cfg.pgSUPassword == "" && cfg.pgSUPasswordFile == "" {
 		die("one of --pg-su-password or --pg-su-passwordfile is required")
 	}
-	if cfg.pgSUPassword != "" && cfg.pgSUPasswordFile != "" {
+	if cfg.pgSUAuthMethod != "trust" && cfg.pgSUPassword != "" && cfg.pgSUPasswordFile != "" {
 		die("only one of --pg-su-password or --pg-su-passwordfile must be provided")
 	}
 
 	if cfg.pgSUUsername == cfg.pgReplUsername {
 		stdout("warning: superuser name and replication user name are the same. Different users are suggested.")
-		if cfg.pgSUPassword != cfg.pgReplPassword {
+		if cfg.pgReplAuthMethod != cfg.pgSUAuthMethod {
+			die("do not support different auth methods when utilizing superuser for replication.")
+		}
+		if cfg.pgSUPassword != cfg.pgReplPassword && cfg.pgSUAuthMethod != "trust" && cfg.pgReplAuthMethod != "trust" {
 			die("provided superuser name and replication user name are the same but provided passwords are different")
 		}
 	}
 
-	if cfg.pgReplPasswordFile != "" {
+	if cfg.pgReplAuthMethod != "trust" && cfg.pgReplPasswordFile != "" {
```

Comment:
The check for `cfg.pgReplAuthMethod != "trust"` can be omitted since it was already checked previously